src/page-ui.js

```diff
@@ -0,0 +1,224 @@
+  const formSection = document.querySelector('#form-section');
+  const newTodoForm = `
+  <section class="section">
+      <div class="container">
+        <div id="todo-form" class="is-hidden">
+
+          <div class="field">
+            <label class="label">Title</label>
+            <div class="control">
+              <input class="input" name="todo-title" type="text" placeholder="Title">
+            </div>
+          </div>
+
+          <div class="field">
+            <label class="label">Description</label>
+            <div class="control">
+              <textarea class="textarea" name="todo-description" placeholder="Description"></textarea>
+            </div>
+          </div>
+
+          <div class="field">
+            <label class="label">Pages</label>
+            <div class="control">
+              <input class="input" name="todo-dueDate" type="date" placeholder="Due date">
+            </div>
+          </div>
+
+          <div class="field">
+            <label class="label">Priority</label>
+            <div class="control">
+              <div class="select">
+                <select name="todo-priority">
+                  <option value="high">High</option>
+                  <option value="normal">Normal</option>
+                  <option value="low">Low</option>
+                </select>
+              </div>
+            </div>
+          </div>
+          
+          <div class="field">
+            <label class="label">Project</label>
+            <div class="control">
+                <div class="select">
+                  <select id="dropdown" name="project-list">
+                  </select>
+                </div>
+            </div>
+          </div>
+
+          <div class="field">
+            <div class="control">
+              <button id="save-todo-button" class="button is-success">Save Todo</button>
+            </div>
+          </div>
+        </div>
+      </div>
+    </section>`;
+  formSection.insertAdjacentHTML('beforeend', newTodoForm);
+}
+
+function generateDropdown() {
+  const select = document.getElementById('dropdown');
+  const list = retrieveProjects();
+  list.forEach(project => {
+    const projectName = project.name;
+    const option = `<option value=${projectName}>${projectName}</option>`;
+    select.insertAdjacentHTML('beforeend', option);
+  });
+}
+
+function saveTodo() {
+  const title = document.querySelector('[name="todo-title"]').value;
+  const description = document.querySelector('[name="todo-description"]').value;
+  const dueDate = document.querySelector('[name="todo-dueDate"]').value;
+  const priority = document.querySelector('[name="todo-priority"]').value;
+  const project = document.querySelector('[name="project-list"]').value;
+
+  const projectList = retrieveProjects();
+
+  const key = projectList.find(element => element.name === project);
+  const todo = new Todo(title, description, dueDate, priority);
+  key.todos.push(todo);
+  console.log('Key: ', key); 
```

Comment:
## Remove unnecessary debugging code 
- [ ] We might have forgotten to remove this debugging code. Debug console statements should be taken out; however, if you're needing to add them to debug a production problem then it may be worth considering if you have enough information being put into your logging framework. Information about parameters, error conditions and so on may well be useful later on when the next bug appears. Using a good logging framework that can behave debug or tracing log messages turned on dynamically can be very useful in the wild.